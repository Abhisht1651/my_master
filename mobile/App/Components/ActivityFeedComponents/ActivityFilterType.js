// @flow

import React, { useState } from 'react'
import { View, TouchableWithoutFeedback, Image, Text } from 'react-native'
import PropTypes from 'prop-types'
import Images from '../../Themes/Images'
import styles from '../Styles/ActivityFeedComponentStyles/ActivityFilterTypeStyles'

// ActivityFilterType is for getting states of each type of Activity (Helper component for ActivityFilter)

type Props = {
  item: {
    name: string,
    id: string
  },
  activityChange: PropTypes.func,
  selected: Array<string>
}

const ActivityFilterType = (props: Props) => {
  const { item, selected } = props
  const [value, setValue] = useState(!!selected.find((e) => e === item.name))
  /* function to show and hide the purple tick */
  const clickHandler = () => {
    setValue(!value)
    props.activityChange(value, props.item.name)
  }

  return (
    <View>
      <TouchableWithoutFeedback onPress={clickHandler}>
        <View>
          <View style={styles.checkbox}>
            {value === true && <Image source={Images.checkedmark} style={styles.checkedBox} />}
          </View>
          <View style={styles.nametext}>
            <Text style={styles.name}>{item.name}</Text>
          </View>
        </View>
      </TouchableWithoutFeedback>
      <View style={styles.border} />
    </View>
  )
}
export default ActivityFilterType
