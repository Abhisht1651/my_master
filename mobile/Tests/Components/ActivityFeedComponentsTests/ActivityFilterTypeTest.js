import React from 'react'
import { TouchableWithoutFeedback } from 'react-native'
import { shallow } from 'enzyme'
import ActivityFilterType from '../../../App/Components/ActivityFeedComponents/ActivityFilterType'
// Test for ActivityFilter
describe('ActivityFilter', () => {
  let wrapper
  it('ActivityFilter component renders correctly', () => {
    wrapper = shallow(<ActivityFilterType selected={[]} item={{ name: 'Wall Post' }} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 TouchableOpacity', () => {
    wrapper = shallow(<ActivityFilterType selected={[]} item={{ name: 'Wall Post' }} />)
    expect(wrapper.find(TouchableWithoutFeedback)).toHaveLength(1)
  })
})
