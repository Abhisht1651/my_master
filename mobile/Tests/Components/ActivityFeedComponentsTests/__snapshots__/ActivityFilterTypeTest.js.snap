// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityFilter ActivityFilter component renders correctly 1`] = `
<View>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#8F8FAF",
            "borderRadius": 30,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "left": 0,
            "marginTop": 20,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "left": 100,
            "marginTop": 30,
            "position": "absolute",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#8E8EAE",
              "fontFamily": "BananaGrotesk-light",
              "fontSize": 28.000000000000004,
              "letterSpacing": 0,
              "lineHeight": 32,
            }
          }
        >
          Wall Post
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>
  <View />
</View>
`;
